codewind-che-sidecar/deploy-pfe/pkg/codewind/util.go

```diff
@@ -0,0 +1,379 @@
+package codewind
+
+import (
+	"encoding/json"
+
+	"k8s.io/client-go/kubernetes"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	extensionsv1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	v1 "github.com/openshift/api/route/v1"
+)
+
+// PatchServiceAccount takes in a list of secret names, and patches it to the specified service account
+func PatchServiceAccount(clientset *kubernetes.Clientset, codewind Codewind) error {
+	patch := ServiceAccountPatch{
+		ImagePullSecrets: &[]ImagePullSecret{
+			{
+				Name: codewind.PullSecret,
+			},
+		},
+	}
+
+	b, err := json.Marshal(patch)
+	if err != nil {
+		return err
+	}
+
+	_, err = clientset.CoreV1().ServiceAccounts(codewind.Namespace).Patch(codewind.ServiceAccountName, types.StrategicMergePatchType, b)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func setPFEEnvVars(codewind Codewind) []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name:  "IN_K8",
+			Value: "true",
+		},
+		{
+			Name:  "PORTAL_HTTPS",
+			Value: "true",
+		},
+		{
+			Name:  "KUBE_NAMESPACE",
+			Value: codewind.Namespace,
+		},
+		{
+			Name:  "TILLER_NAMESPACE",
+			Value: codewind.Namespace,
+		},
+		{
+			Name:  "CHE_WORKSPACE_ID",
+			Value: codewind.WorkspaceID,
+		},
+		{
+			Name:  "PVC_NAME",
+			Value: codewind.PVCName,
+		},
+		{
+			Name:  "SERVICE_NAME",
+			Value: "codewind-" + codewind.WorkspaceID,
+		},
+		{
+			Name:  "SERVICE_NAME",
+			Value: "codewind-" + codewind.WorkspaceID,
+		},
+		{
+			Name:  "SERVICE_ACCOUNT_NAME",
+			Value: codewind.ServiceAccountName,
+		},
+		{
+			Name:  "MICROCLIMATE_RELEASE_NAME",
+			Value: "RELEASE-NAME",
+		},
+		{
+			Name:  "HOST_WORKSPACE_DIRECTORY",
+			Value: "/projects",
+		},
+		{
+			Name:  "CONTAINER_WORKSPACE_DIRECTORY",
+			Value: "/codewind-workspace",
+		},
+		{
+			Name:  "OWNER_REF_NAME",
+			Value: codewind.OwnerReferenceName,
+		},
+		{
+			Name:  "OWNER_REF_UID",
+			Value: string(codewind.OwnerReferenceUID),
+		},
+		{
+			Name:  "CODEWIND_PERFORMANCE_SERVICE",
+			Value: codewind.PerformanceName + "-" + codewind.WorkspaceID,
+		},
+		{
+			Name:  "CHE_INGRESS_HOST",
+			Value: codewind.Ingress,
+		},
+	}
+}
+
+func setPerformanceEnvVars(codewind Codewind) []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name:  "IN_K8",
+			Value: "true",
+		},
+		{
+			Name:  "PORTAL_HTTPS",
+			Value: "true",
```

Comment:
This was previously false, just making sure we're intentionally making it true now.

Reply:
Good catch, set it back to false.